core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -129,14 +130,12 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
         if (queryBuilder != null) {
             searchSourceBuilder.query(queryBuilder);
         }
-
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        if (request.hasParam("from")) {
+            searchSourceBuilder.from(request.paramAsInt("from", DEFAULT_VALUE));
```

Comment:
This is also technically a breaking change so it can only be applied to 6.0 and will need an entry in the breaking changes documentation. It is a fine change to make though.